src/Sentry/SentryOptions.cs

```diff
@@ -542,8 +542,19 @@ public int MaxCacheItems
 
     /// <summary>
     /// Creates the inner most <see cref="HttpClientHandler"/>.
+    /// Deprecated in favor of <see cref="CreateHttpMessageHandler"/>.
     /// </summary>
-    public Func<HttpClientHandler>? CreateHttpClientHandler { get; set; }
+    [Obsolete("Use CreateHttpMessageHandler instead")]
+    public Func<HttpClientHandler>? CreateHttpClientHandler
+    {
+        get => CreateHttpMessageHandler as Func<HttpClientHandler>;
```

Comment:
Does this typecast work? If not, it'll just return `null`

Reply:
Unfortunately you are right. This doesn't work. I introduced a private field to hold the `Func<HttpClientHandler>?`. So when the setter is used, the getter will work. if the new `CreateHttpMessageHandler` is used instead `CreateHttpClientHandler`will return null.